controllers/apps/operations/backup.go

```diff
@@ -0,0 +1,179 @@
+)
+
+type BackupOpsHandler struct{}
+
+var _ OpsHandler = BackupOpsHandler{}
+
+func init() {
+	// ToClusterPhase is not defined, because 'backup' does not affect the cluster phase.
+	backupBehaviour := OpsBehaviour{
+		FromClusterPhases:                  appsv1alpha1.GetClusterUpRunningPhases(),
+		OpsHandler:                         BackupOpsHandler{},
+		ProcessingReasonInClusterCondition: ProcessingReasonBackup,
+	}
+
+	opsMgr := GetOpsManager()
+	opsMgr.RegisterOps(appsv1alpha1.BackupType, backupBehaviour)
+}
+
+// ActionStartedCondition the started condition when handling the backup request.
+func (b BackupOpsHandler) ActionStartedCondition(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) (*metav1.Condition, error) {
+	return appsv1alpha1.NewBackupCondition(opsRes.OpsRequest), nil
+}
+
+// Action implements the backup action.
+// It will create a backup resource for cluster.
+func (b BackupOpsHandler) Action(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) error {
+	opsRequest := opsRes.OpsRequest
+	cluster := opsRes.Cluster
+
+	// create backup
+	if backup, err := buildBackup(reqCtx, cli, opsRequest, cluster); err != nil {
+		return err
+	} else {
+		return cli.Create(reqCtx.Ctx, backup)
+	}
+}
+
+// ReconcileAction implements the backup reconcile action.
+// It will check the backup status and update the OpsRequest status.
+// If the backup is completed, it will return OpsSuccess
+// If the backup is failed, it will return OpsFailed
+func (b BackupOpsHandler) ReconcileAction(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) (appsv1alpha1.OpsPhase, time.Duration, error) {
+	opsRequest := opsRes.OpsRequest
+	cluster := opsRes.Cluster
+
+	// get backup
+	backups := &dataprotectionv1alpha1.BackupList{}
+	if err := cli.List(reqCtx.Ctx, backups, client.InNamespace(cluster.Namespace), client.MatchingLabels(getBackupLabels(cluster.Name, opsRequest.Name))); err != nil {
+		return appsv1alpha1.OpsFailedPhase, 0, err
+	}
+
+	if len(backups.Items) == 0 {
+		return appsv1alpha1.OpsFailedPhase, 0, fmt.Errorf("backup not found")
+	}
+	// check backup status
+	phase := backups.Items[0].Status.Phase
+	if phase == dataprotectionv1alpha1.BackupCompleted {
+		return appsv1alpha1.OpsSucceedPhase, 0, nil
+	} else if phase == dataprotectionv1alpha1.BackupFailed {
+		return appsv1alpha1.OpsFailedPhase, 0, fmt.Errorf("backup failed")
+	}
+	return appsv1alpha1.OpsRunningPhase, time.Second, nil
+}
+
+// SaveLastConfiguration records last configuration to the OpsRequest.status.lastConfiguration
+func (b BackupOpsHandler) SaveLastConfiguration(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) error {
+	return nil
+}
+
+// GetRealAffectedComponentMap gets the real affected component map for the operation
+func (b BackupOpsHandler) GetRealAffectedComponentMap(opsRequest *appsv1alpha1.OpsRequest) realAffectedComponentMap {
```

Comment:
this function has been removed.

Reply:
https://github.com/apecloud/kubeblocks/blob/ea6bfc6b83690ee9863358cc030b4f2938c601d4/controllers/apps/operations/backup.go#L40

the `BackupHandler` must have `GetRealAffectedComponentMap`.

https://github.com/apecloud/kubeblocks/blob/0733806e4b39ffc1f4ee95b28e4bff8d8901f5f9/controllers/apps/operations/type.go#L33-L49

Reply:
rebase from main